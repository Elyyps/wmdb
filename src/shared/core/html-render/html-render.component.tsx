import * as React from "react";
import { Parser } from "html-to-react";

export interface IHtmlRenderComponentProps {
  html: string;
}

const HtmlRenderComponent = (props: IHtmlRenderComponentProps) => (
  <React.Fragment>{Parser().parse(props.html)} </React.Fragment>
);

export { HtmlRenderComponent };
