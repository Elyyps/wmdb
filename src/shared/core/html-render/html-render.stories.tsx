import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { HtmlRenderComponent } from "./html-render.component";

storiesOf("HtmlRender", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => <HtmlRenderComponent html="<h1>HTML rendered</h1><br/>From a string/>" />);
