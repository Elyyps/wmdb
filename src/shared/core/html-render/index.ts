export * from "./html-render.component";
