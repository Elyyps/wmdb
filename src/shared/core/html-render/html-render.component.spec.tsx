import React from "react";
import { shallow } from "enzyme";
import { HtmlRenderComponent } from "./html-render.component";

describe("[HtmlRender]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<HtmlRenderComponent html="<h1>HTML rendered</h1><br/>From a string/>" />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
