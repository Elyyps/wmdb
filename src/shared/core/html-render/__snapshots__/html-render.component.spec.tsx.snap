// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[HtmlRender] should render component without crashing 1`] = `
<Fragment>
  <h1
    key="0"
  >
    HTML rendered
  </h1>
  <br
    key="1"
  />
  From a string/&gt;
   
</Fragment>
`;
